### PR TITLE
Implements ATF tests for iAP2 Bluetooth to USB transport switching

### PR DESCRIPTION
**Pull request by dev-gh**
This pull requests includes:
- transport switching happy path test w/o data to resume
- transport switching in case application has some data to resume (basic)
- transport switching in case application has data to resume (full data set and checks)
- commands holding during transport switching
- common helper for tests

**Update**: tests cover prior implementation of switching when it was done automatically on SDL side. Now SDL implementation is changed with new sequence so tests must be updated and new iAP emulation must be implemented on SDL side.

### CODE REVIEW

File: test_scripts/smoke_api.lua [hunk 1]
```diff
@@ -514,7 +514,7 @@ end
 
 -- 2. PutFiles
 	commonSteps:PutFile("PutFile_action.png", "action.png")
-	commonSteps:PutFile("PutFile_MaxLength_255Characters", strMaxLengthFileName255)
+	commonSteps:PutFile("PutFile_MaxLength_255Characters", strMaxLengthFileName242)
```
Comment (jacobkeeler) [inline test_scripts/smoke_api.lua]: `PutFile_MaxLength_255Characters` to `PutFile_MaxLength_242Characters `

File: test_scripts/iAP2TransportSwitch/002_resume_failed_simple_case.lua [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+          sessionBluetooth:ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+        end)
+      EXPECT_HMICALL("BasicCommunication.UpdateAppList")
+      :Do(function(_, data)
+          self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", { })
+        end)
+      EXPECT_HMINOTIFICATION("Buttons.OnButtonSubscription", {
+        isSubscribed = true,
+        name = "CUSTOM_BUTTON"
+      })
```
Comment (jacobkeeler) [inline test_scripts/iAP2TransportSwitch/002_resume_failed_simple_case.lua]: Why is this subscription expected? I don't see any corresponding `SubscribeButton` RPC sent from the mobile session

File: test_scripts/Smoke/HeartBeat/ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua [hunk 3]
```diff
@@ -115,4 +117,4 @@ function Test.Stop_SDL()
   StopSDL()
 end
 
-return Test
\ No newline at end of file
+return Test
```
Comment (jacobkeeler) [inline test_scripts/Smoke/HeartBeat/ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua]: Smoke test changes seem out of scope, any reason they are included in this PR?

File: test_scripts/iAP2TransportSwitch/003_resume_failed_remove_data_check.lua [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+        self:FailTestCase("File '" .. pFileName .. "' is not found")
+      end
+   end)
+end
+
+local function connectBluetoothDevice(self)
+  deviceBluetooth = self:createIAP2Device(common.device.bluetooth.id,
+    common.device.bluetooth.port, common.device.bluetooth.out)
+
+  EXPECT_HMICALL("BasicCommunication.UpdateDeviceList", {
```
[the remark this answers is not in the document]
Comment (dboltovskyi) [reply test_scripts/iAP2TransportSwitch/003_resume_failed_remove_data_check.lua]: @jacobkeeler Unfortunately we don't have expectation functions (like ```expectCall```) for HMI connection. Currently there are only ```send``` ones available.
Expectation functions are going to be implemented in scope of [ATF Facade](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0121-atf_facade_proposal.md) proposal

File: test_scripts/iAP2TransportSwitch/002_resume_failed_simple_case.lua (hunk 2 above)
Comment (dboltovskyi) [reply test_scripts/iAP2TransportSwitch/002_resume_failed_simple_case.lua]: @jacobkeeler This event is expected due to SDL requirement:

> By default, SDL must subscribe every application to CUSTOM_BUTTON right after the application's registration ONLY if system supports CUSTOM_BUTTON

File: test_scripts/Smoke/HeartBeat/ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua (hunk 3 above)
Comment (dboltovskyi) [reply test_scripts/Smoke/HeartBeat/ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua]: @jacobkeeler Actually changes for smoke should not be shown by GitHub since they are already in develop. For example please take for these 2 versions of ```ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua``` script:
[develop](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Smoke/HeartBeat/ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua)
[feature](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/feature/iap2_transport_switch_tests/test_scripts/Smoke/HeartBeat/ATF_HeartBeat_App_does_not_send_HB_and_does_not_respond.lua) 

They are identical.

This issue can be resolved by re-basing feature branch on current develop.
If you don't mind I'll do this.

File: test_scripts/smoke_api.lua (hunk 1 above)
Comment (dboltovskyi) [reply test_scripts/smoke_api.lua]: @jacobkeeler Change won't be visible after re-base on develop

Comment (dboltovskyi) [reply test_scripts/smoke_api.lua]: @jacobkeeler Branch is re-based on current develop

**Comment by dboltovskyi** [comment]
Smoke tests were updated in 792f1c3 due to changes in SDL Core introduced in https://github.com/smartdevicelink/sdl_core/commit/82d13ffda5f1d946f815a9f0a7de13e0add2fc44
